packages/shared/src/variants/quantum.ts

```diff
@@ -0,0 +1,218 @@
+  constructor(config: BadukConfig) {
+    super(config);
+
+    this.subgames = [];
+  }
+
+  playMove(player: number, move: string): void {
+    if (move === "resign") {
+      this.phase = "gameover";
+      this.result = player === 0 ? "B+R" : "W+R";
+      return;
+    }
+
+    if (move === "timeout") {
+      this.phase = "gameover";
+      this.result = player === 0 ? "B+T" : "W+T";
+      return;
+    }
+
+    const pos = Coordinate.fromSgfRepr(move);
+    // TODO: add a Dimensions class (#216) and look at that instead of
+    // building a grid for no reason
+    if (!new Grid(this.config.width, this.config.height).isInBounds(pos)) {
+      throw "Out of bounds!";
+    }
+
+    switch (this.round) {
+      case 0: {
+        if (player !== 0) {
+          throw new Error("Black must place the first quantum stone.");
+        }
+        this.first_quantum_stone = pos;
+        break;
+      }
+      case 1: {
+        if (player !== 1) {
+          throw new Error("White must place the second quantum stone.");
+        }
+        this.subgames = [
+          new QuantumSubgame(this.config),
+          new QuantumSubgame(this.config),
+        ];
+        // We can assume this is non-null because it is filled in the first round.
+        const first = this.first_quantum_stone!;
+        this.subgames[0].addQuantumStone(first, pos, Color.BLACK);
+        this.subgames[0].addQuantumStone(pos, first, Color.WHITE);
+        this.subgames[1].addQuantumStone(pos, first, Color.BLACK);
+        this.subgames[1].addQuantumStone(first, pos, Color.WHITE);
+        break;
+      }
+      default: {
+        const quantum_captures = this.subgames.map(
+          (game) => game.play(player, move).quantum_mapped_captures,
+        );
+        this.subgames[0].clear(quantum_captures[1]);
+        this.subgames[1].clear(quantum_captures[0]);
+      }
+    }
+
+    super.increaseRound();
+  }
+
+  exportState(): QuantumGoState {
+    if (!this.subgames.length) {
+      const makeBoardWithStone = (
+        pos: Coordinate | undefined,
+        color: Color,
+      ) => {
+        const board = new Grid<Color>(
+          this.config.width,
+          this.config.height,
+        ).fill(Color.EMPTY);
+        if (pos != null) {
+          board.set(pos, color);
+        }
+        return board.to2DArray();
+      };
+      const pos = this.first_quantum_stone?.toSgfRepr() ?? null;
+
+      return {
+        boards: [
+          makeBoardWithStone(this.first_quantum_stone, Color.BLACK),
+          makeBoardWithStone(this.first_quantum_stone, Color.WHITE),
+        ],
+        quantum_stones: [
+          [pos, null],
+          [null, pos],
+        ],
+      };
+    }
+
+    return {
+      boards: this.subgames.map((game) => game.badukGame.board.to2DArray()),
+      quantum_stones: this.subgames[0].quantumMappings.map((mapping) =>
+        mapping.map((pos) => pos.toSgfRepr()),
+      ),
+    };
+  }
+  nextToPlay(): number[] {
+    return [this.round % 2];
```

Comment:
Please return empty array when the game is finished.

Reply:
Will do!

And whoops... I am realizing I forgot to implement endgame.  Sending back to draft

Reply:
Fixed!